### ISSUE TITLE
What is the pre-training objective to the model bart-base

### ISSUE THREAD

**Issue by chris4540** [pytorch/fairseq#2303]
### Before asking:   
1. search the issues.   
2. search the docs.    

#### What is your question?
What is the pre-training objective to the model: `bart-base`

I cannot find the pre-training objective either in the README file or the paper.

#### Code
N/A

#### What have you tried?
N/A

#### What's your environment?

N/A

**Comment by lematt1991** [comment]
From section 2.2 of the paper:

> BART is trained by corrupting documents and then optimizing a reconstruction loss—the cross-entropy between the decoder’s output and the original document.
Unlike existing denoising autoencoders, which are tailored to specific noising schemes, BART allows us to
apply any type of document corruption. In the extreme
case, where all information about the source is lost,
BART is equivalent to a language model.

In fairseq, it uses the `denoising` task.

**Comment by chris4540** [comment]
Thanks for your reply.

Let me clarify my question again.

In section 5.1,
> Based on the results in Section 4, we use a combination of text infilling and sentence permutation.

Therefore, I can see that the pretrained `bart-large` used the objective above to train yet it does not state clearly in the [README](https://github.com/pytorch/fairseq/blob/master/examples/bart/README.md) file. 

I have checked the `denosing` task from `fairseq/tasks/denoising.py`. Is the bart-base trained with this file and using the default arguments?

**Comment by lematt1991** [comment]
If you download the checkpoint, you can load it and inspect the arguments that were used to train it.  For example:

```python
import torch
chkpnt = torch.load('model.pt')
args = chkpnt['args']
```

[Here](https://gist.github.com/lematt1991/e72c01145c0a9fc0618dc147b4f2a4b4) is a gist containing a JSON dump of the args used to train `bart.base`

**Comment by chris4540** [comment]
Thx, I will study the files you provided.
